### DISCUSSION TITLE
BirdnetPi vs iOS app

### DISCUSSION THREAD

**Discussion by ExperiMentor** [mcguirepr89/BirdNET-Pi#685] [Q&A] [answered]
I have a Pi 02W running BirdnetPi and an iPad running the analagous Birdweather app. They have identical microphones located a couple of feet apart. But the iOS app is detecting around 20 times more birdsongs every day, consistently. It's the same if I switch the microphones over.
Different 'Settings' are available on the 2 systems, but I have adjusted them as similarly as I can see how (eg confidence %age).
I don't expect identical results, but this difference seems huge. What is wrong?
What are the suggested 'Settings' for BirdnetPi?

**Comment by Fish31520** [comment, marked as answer]
Hi,

3 months ago, @ehpersonal38 replied this, it might interest you : 

_"I do suspect that the low confidence behavior you're seeing is due to the fact that BirdNET-Pi is currently using a model that was last updated about 2 years ago whereas the BirdNET mobile app (and I'm sure Merlin too) gets regular model updates. Patrick has begun work as of last month on implementing the new model, though I suspect it won't be fully ready for another several months."_

**Reply by mcguirepr89** [reply]
This is mostly correct -- BirdNET-Pi is the only BirdNET iteration of which I know that uses the [TFLite model](https://github.com/kahst/BirdNET-Lite). It is deprecated in favor of [the newer models](https://github.com/kahst/BirdNET-Analyzer), according to its author, Stefan Kahl (@kahst).

I have discontinued, however, the development for the newer model, so you should not ever expect that from me.

**Comment by jflyons** [comment]
Hi @ExperiMentor - I have nearly the same dual setup as you describe. I have over the past weeks been using the Birdweather app on my fairly old iPad, no external microphone, letting it run during daylight hours (which even in freezing temps goes a full day without a power recharge). It picks up many more detections than does my BirdNET-Pi, with a mic in an opposite side yard. Your more direct comparison is very interesting, as mine's a bit apples and oranges. Overflying geese, ducks, gulls and corvids can be detected on both, but not the land/garden birds. I received great advice particularly from @ehpersonal38 on adjusting the internal settings in BirdNET-Pi to species-specific (and lower than .7) levels. But like you, I wonder what else I can do to understand their respective performance.

Overall, I must add that BirdNET-Pi starting mid-2022 and Birdweather later, have been fantastic tools in helping me pursue a life-long quest of knowing my birds!

**Reply by ExperiMentor** [reply]
@jflyons Thanks for getting in touch. I understood the explanation to be that the TFlite model used in BirdNET-Pi has been deprecated as it performs much less well than the full TensorFlow model that is now available and in use on other platforms. I guess a Raspberry Pi isn't felt to be capable of adequately supporing the more intensive algorithms. So I discontinued my BirdNET-Pi. The 'promotional' articles in RasPiMag etc shoudl have made clear that the project was interesting but now deprecated.

My setup is now an old iPad (I hate them so was pleased to find a use for it), with a headphone/power Y-splitter attached so that I can use an external microphone (ie old Apple lightning earpieces inside a plastic bag hung out a bedroom window!) and supply power from USB to lightning at the same time for continuous 24/7 running. This setup has now reached 75,000 bird calls recognised (and on the trace there are at least as many again that are not identified). I am [https://app.birdweather.com/stations/604](url)

**Reply by jflyons** [reply]
Hi @ExperiMentor - Thanks for response and interesting info! That's a helpful explanation of the current state of BirdNET-Pi and I will say for me as a lifelong birder (and already feeling the Merlin sound app has been the learning help I'd been wanting for decades), its performance on true positives is quite good, and the false positives can be instructive too! I also like the "excluded species" feature for helping filter, though just in the past couple of days I have had a few way-out-of-season birds that slip through the model, I guess. Its performance on the missed birds, false negatives, leads to the ratios in the comparisons you report and I observe too. These are birds I can hear - often through a window - but don't reach the .7 threshold for the model and I've hesitated to do the custom (lower) thresholds, as kindly suggested.)

Your iPad setup sounds very practical and the way I like to do things too - the plastic bag out the window is exactly my style! I checked out your station - very impressive diversity of species! - and invite you to see mine - daytime hours only, for now :)
[[https://app.birdweather.com/stations/604](https://github.com/mcguirepr89/BirdNET-Pi/discussions/url)](url)

**Reply by nilspupils** [reply]
@ExperiMentor: Please have a look at this comment for clarification: 
https://github.com/mcguirepr89/BirdNET-Pi/discussions/69#discussioncomment-4593907

**Reply by ehpersonal38** [reply]
If you'd like to test the new model please see this: https://github.com/mcguirepr89/BirdNET-Pi/discussions/184#discussioncomment-4691491

**Reply by DD4WH** [reply]
@ExperiMentor 
The "number of birdsongs detected" is surely not a good indicator of the quality of a classifier. Did you have a look how many of them are false positives ?
And it depends on the aim you would like to achieve with your system, e.g. how many false positives are acceptable for your aim?

Your statement "the TFlite model used in BirdNET-Pi has been deprecated as it performs much less well than the full TensorFlow model" is surely wrong, as is shown in this paper: https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.14003
Have a look at figure 1.

The testing of the quality of different classifiers is very difficult and data- and time-consuming. So I would be very careful about statements on one model being much less performant as another model.

**Reply by ExperiMentor** [reply]
I agree that classification of relative performance is difficult. My point was that the 2 systems, promoted as being similar, give radically different results. Going that far is easy.
The 'deprecated' quote is from the BirdNET-Pi  GitHub - ie it's own developers are saying 'should no longer be used'.

**Reply by ExperiMentor** [reply]
I haven't attempted any formal performance analysis. But the 20-fold difference in frequency applies across the board, including the 'Top 5' which are appropriate and correct species near me (eg more than half on both systems are European Robin, which are prevalent here at this time of year). And yet one system reports 20x more of them than the other, even when same confidence selected.
